### PR TITLE
Add analysis and emit difference time spans to EnC telemetry

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

File: src/Features/Core/Portable/EditAndContinue/TraceLog.cs [hunk 1]
```diff
@@ -89,7 +89,7 @@ private Arg(int value, StrongBox<EnumType> enumKind)
             public static implicit operator Arg(bool value) => new(value ? "true" : "false");
             public static implicit operator Arg(ProjectId value) => new(value.DebugName);
             public static implicit operator Arg(DocumentId value) => new(value.DebugName);
-            public static implicit operator Arg(Diagnostic value) => new(value);
+            public static implicit operator Arg(Diagnostic value) => new(value.ToString());
```
Comment (tmat) [inline src/Features/Core/Portable/EditAndContinue/TraceLog.cs:92]: Avoid holding on Diagnostic instance from the log - it may hold on SyntaxTree.

**Comment by tmat** [comment]
@phil-allen-msft

**Comment by tmat** [comment]
@davidwengier ptal

File: src/Features/Core/Portable/EditAndContinue/DebuggingSessionTelemetry.cs [hunk 2]
```diff
@@ -115,6 +115,15 @@ public static void Log(Data data, Action<FunctionId, LogMessage> log, Func<int>
...
                     map["ProjectIdsWithAppliedChanges"] = editSessionData.Committed ? editSessionData.ProjectsWithValidDelta.Select(ProjectIdToPii) : "";
+
+                    // Total milliseconds it took to emit the delta in this edit session.
+                    map["EmitDifferenceMilliseconds"] = editSessionData.EmitDifferenceTime.Milliseconds;
+
+                    // Total milliseconds it took to analyze all documents that contributed to the changes that were
+                    // attempted to be applied (whether or not the applications was successful) in this edit session.
+                    // Includes analysis that had been performed asynchronously before "apply changes" was triggered
+                    // (if we reused analysis results that were calculated by EnC analyzer for rude edit reporting).
+                    map["TotalAnalysisMilliseconds"] = editSessionData.AnalysisTime.Milliseconds;
```
Comment (davidwengier) [inline src/Features/Core/Portable/EditAndContinue/DebuggingSessionTelemetry.cs:126]: Should be `TotalMilliseconds`. `Milliseconds` is just the ms component of whatever the time is. Also applies to line 120

**Review by davidwengier — APPROVED** [review]
Going to approve so you're not blocked on me, because its an otherwise straight forward PR and I trust you to make the updates, but to be very clear: As written this will report incorrect telemetry.

File: src/Features/Core/Portable/EditAndContinue/DebuggingSessionTelemetry.cs (hunk 2 above)
Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/DebuggingSessionTelemetry.cs:126]: Good catch. Thanks!

**Pull request merged**